### PR TITLE
Suggestion for #72

### PR DESCRIPTION
**Pull request by paul-asvb**
I have a simple draft for the first steps of my suggestion

### CODE REVIEW

**Comment by ASVBPREAUBV** [comment]
@samuell What do you think? I will keep working on it if you like it.

**Comment by samuell** [comment]
@ASVBPREAUBV Oh, sorry, I seem to have missed that you sent a PR already! I'll have a deeper look shortly, adding some shorter comments now.

**Comment by samuell** [comment]
Btw, I need to update the CircleCI config for CircleCI 2.0 as well :P

File: workflow.go [hunk 1]
```diff
@@ -57,6 +58,24 @@ type WorkflowProcess interface {
...
 // ----------------------------------------------------------------------------
 
+// NewWorkflow returns a new Workflow
+func NewWorkflowWithPath(name string, maxConcurrentTasks int, workflowPath string) *Workflow {
+	wf := newWorkflowWithoutLogging(name, maxConcurrentTasks)
+
+	// Set up logging
+	allowedCharsPtrn := regexp.MustCompile("[^a-z0-9_]")
+	wfNameNormalized := allowedCharsPtrn.ReplaceAllString(strings.ToLower(name), "-")
+	logFilePathRelative := fmt.Sprintf("log/scipipe-%s-%s.log", time.Now().Format("20060102-150405"), wfNameNormalized)
```
[the remark this answers is not in the document]
Comment (ASVBPREAUBV) [reply workflow.go]: I thought this is the point. To include the output of the executed commands the output. Or is that bad practice? would you want to separate that?

File: workflow.go [hunk 2]
```diff
@@ -57,6 +58,24 @@ type WorkflowProcess interface {
 // Factory function(s)
 // ----------------------------------------------------------------------------
 
+// NewWorkflow returns a new Workflow
+func NewWorkflowWithPath(name string, maxConcurrentTasks int, workflowPath string) *Workflow {
```
[the remark this answers is not in the document]
Comment (samuell) [reply workflow.go]: Perhaps `NewWorkflowWithLogPath()` is best. Do you want to update, or should I amend?

Btw, another nitpick: Preferable if docstrings are broken at 80 chars limit (I use to format with `g`,`q` commands in vim :)

[the remark this answers is not in the document]
Comment (ASVBPREAUBV) [reply workflow.go]: Easy, ill do it

**Comment by samuell** [comment]
@ASVBPREAUBV could you please rebase towards the updated scipipe develop or master now, after I updated the CircleCI 2.0 ... so the unit tests can run properly?

File: workflow.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (samuell) [reply workflow.go]: > I thought this is the point. To include the output of the executed commands the output. Or is that bad practice? would you want to separate that?

What I mean is that if `log/` is included in the relative path, which will always be added to the log path, then you can't specify that you want your log files to end up in a folder called `/some/path/scipipe-logs/` ... but will always have this extra `log/` part in the end, so in this case: `/some/path/scipipe-logs/log`.

[the remark this answers is not in the document]
Comment (samuell) [reply workflow.go]: @ASVBPREAUBV But aha, so the idea was to also make all file outputs go to that directory?

I think that is definitely a separate issue, and might have more far-reaching consequences though and would need a separate mechanism, such as specifying "executionDirectory". So I don't think it should be included here.

**Comment by samuell** [comment]
@ASVBPREAUBV I think there is here a confusion about two issues.

1. #72 is about being able to refer to the conrete path of temporary output folders for tasks.

2. The changes proposed in this PR are instead referring to changing the application log path, and would not change the temporary dirs mentioned in #72 at all.

I'm thus a bit unclear about what you wanted to solve in the first place. I guess it is something more related to issue #72? Let's perhaps discuss that further in that issue? As it is now, this PR does not address that issue, AFAIS.

File: workflow.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (ASVBPREAUBV) [reply workflow.go]: i agree with you. i was thinking of a IP type e.g.: file, directory...

**Comment by ASVBPREAUBV** [comment]
Sorry for the confusion. You can just close this issue. Ill cherry-pick the changes for https://github.com/scipipe/scipipe/issues/72 from my changes.

**Comment by samuell** [comment]
@paul-asvb Not sure if this was ever resolved (my memory is failing me a bit ...)? Closing for now, but please re-open if you think it is not yet resolved.

**Pull request closed without merging**